Updates to pfcwd tests to handle new behaviour of cisco-8000.

rraghav-cisco opened the pull request:
### Description of PR
Recently cisco-8000 has changed the behaviour of pfcwd: It will not trigger watchdog in low/zero traffic conditions, even if it is receiving pfc storm. This new behaviour is in contrast to the old behaviour - which is trigger watchdog when there is a pfc storm, even if there is no traffic. The existing pfcwd tests are written to test this behaviour. This PR attempts to update the scripts based on new behaviour, only for cisco-8000. 

In this PR:
1. Skip pfcwd/test_pfcwd_timer_accuracy.py for cisco-8000
2. Update pfcwd/test_pfcwd_function.py for cisco-8000's new behaviour.
3. The timer_accuracy is being replaced by another script for cisco-8000: https://github.com/sonic-net/sonic-mgmt/pull/11747

Summary:
Fixes pfcwd script failures due to new cisco-8000 behaviour.

### Type of change

- [X] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)


### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [X] 202305
- [X] 202311

### Approach
#### What is the motivation for this PR?
Pls see the description.
#### How did you do it?
We will remove the expected log analyzer messages for pfcwd, since it is not triggered anymore. Also skip timer_accuracy, since it is completely dependent on it.

#### How did you verify/test it?
Yet to be verfied on my TB.

#### Any platform specific information?
Applies to only cisco-8000 for now.

alpeshspatel reviewed (approved):
for cisco-8000 due to the new behavior to not trigger WD with no traffic

lipxu commented on tests/pfcwd/test_pfcwd_all_port_storm.py:
```diff
@@ -156,12 +156,23 @@ def test_all_port_storm_restore(self, duthosts, enum_rand_one_per_hwsku_frontend
+                storm_hndle,
+                expect_regex=[fetch_vendor_specific_diagnosis_re(duthost)],
+                syslog_marker="all_port_storm",
+                action="storm")
```
Why do we need to run the storm test here? how to check the result? what's the difference between pass and failure?

rraghav-cisco replied:
@lipxu : We makes sure that the storm is *not* detected. By removing EXPECT_PFC_WD_DETECT_RE, the loganalyzer will flag an error if the DUT pfcwd flags a storm.

lipxu replied:
Not sure how loganalyzer flag an error if the DUT pfcwd flags a storm here?

lipxu commented on tests/pfcwd/test_pfcwd_function.py:
```diff
@@ -263,6 +263,8 @@ def get_pkt_cnts(self, queue_oid, begin=True):
 
         """
+        if self.dut.facts["asic_type"] in ['cisco-8000']:
+            return
```
If return here, the cntr_val would not be updated? will it be failed during verify_pkt_cnts?

rraghav-cisco commented:
@kevinskwang , @lipxu : I have addressed the comments, and made sure it works:

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---------------------------------------------------------------------------- generated xml file: /run_logs/logs/pfcwd/tr_2024-03-08-23-29-54.xml -----------------------------------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
--------------------------------------------------------------------------------------------------- live log sessionfinish ---------------------------------------------------------------------------------------------------
23:36:07 __init__.pytest_terminal_summary         L0064 INFO   | Can not get Allure report URL. Please check logs
================================================================================================== short test summary info ===================================================================================================
XFAIL pfcwd/test_pfcwd_function.py::TestPfcwdFunc::test_pfcwd_actions[mth64-m5-2]
  The test is using pfc_gen.py, which leaks out the packets.
XFAIL pfcwd/test_pfcwd_function.py::TestPfcwdFunc::test_pfcwd_multi_port[mth64-m5-2]
  The test is using pfc_gen.py, which leaks out the packets.
XFAIL pfcwd/test_pfcwd_function.py::TestPfcwdFunc::test_pfcwd_mmu_change[mth64-m5-2]
  The test is using pfc_gen.py, which leaks out the packets.
XFAIL pfcwd/test_pfcwd_function.py::TestPfcwdFunc::test_pfcwd_port_toggle[mth64-m5-2]
  The test is using pfc_gen.py, which leaks out the packets.
========================================================================================= 4 xfailed, 1 warning in 371.70s (0:06:11) ==========================================================================================
sonic@447c4c4ec6fc:/data/tests$ 

```

rraghav-cisco commented:
```
=========================================================================================================== PASSES ===========================================================================================================
_______________________________________________________________________________ TestPfcwdAllPortStorm.test_all_port_storm_restore[mth64-m5-2] ________________________________________________________________________________
---------------------------------------------------------------------------- generated xml file: /run_logs/logs/pfcwd/tr_2024-03-09-00-31-30.xml -----------------------------------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
--------------------------------------------------------------------------------------------------- live log sessionfinish ---------------------------------------------------------------------------------------------------
00:35:49 __init__.pytest_terminal_summary         L0064 INFO   | Can not get Allure report URL. Please check logs
================================================================================================== short test summary info ===================================================================================================
PASSED pfcwd/test_pfcwd_all_port_storm.py::TestPfcwdAllPortStorm::test_all_port_storm_restore[mth64-m5-2]
========================================================================================== 1 passed, 1 warning in 257.97s (0:04:17) ==========================================================================================
sonic@447c4c4ec6fc:/data/tests$ 

```

kevinskwang commented on tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:
```diff
@@ -917,6 +917,18 @@ pfcwd/test_pfc_config.py::TestPfcConfig::test_forward_action_cfg:
 
+pfcwd/test_pfcwd_function.py:
+   xfail:
+     reason: "The test is using pfc_gen.py, which leaks out the packets."
```
we usually used pfc_gen.py at there, why it started to fail and need  to skip recently?

kevinskwang commented on tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:
```diff
@@ -917,6 +917,18 @@ pfcwd/test_pfc_config.py::TestPfcConfig::test_forward_action_cfg:
+
+pfcwd/test_pfcwd_timer_accuracy.py:
+   skip:
+     reason: "The test is replaced by pfcwd/test_pfcwd_timer_accuracy_cisco_8000.py"
```
What's the PR for test_pfcwd_timer_accuracy_cisco_8000.py?

kevinskwang commented on tests/pfcwd/test_pfcwd_function.py:
```diff
@@ -1115,6 +1145,9 @@ def test_pfcwd_port_toggle(self, request, fake_storm, setup_pfc_test, setup_dut_
 
+            except RunAnsibleModuleFail:
+                if duthost.facts['asic_type'] in ['cisco-8000']:
+                    request.applymarker(pytest.mark.xfail(run=True))
```
doesn't make sense to me to mark as xfail here

kevinskwang commented on tests/pfcwd/test_pfcwd_function.py:
```diff
@@ -1105,8 +1126,17 @@ def test_pfcwd_port_toggle(self, request, fake_storm, setup_pfc_test, setup_dut_
-                loganalyzer.expect_regex.extend([EXPECT_PFC_WD_DETECT_RE + fetch_vendor_specific_diagnosis_re(duthost)])
-                loganalyzer.match_regex = []
+                if duthost.facts["asic_type"] in ['cisco-8000']:
+                    loganalyzer.expect_regex.extend(
```
Can I understand you just remove the pfcwd storm detect/restore syslog checking here?

rraghav-cisco commented:
As per discussions with @kevinskwang and @XuChen-MSFT :
1. This PR will be closed.
2. A new set of scripts which first start a continuous traffic stream and then initiate the pfcwd tests will be written, specifically for cisco-8000.
